Content.Server/NPC/Systems/NPCSystem.cs

```diff
@@ -55,6 +57,8 @@ public void OnPlayerNPCDetach(EntityUid uid, HTNComponent component, PlayerDetac
             if (TryComp<MindContainerComponent>(uid, out var mindContainer) && mindContainer.HasMind)
                 return;
 
+            RemComp<MouseRotatorComponent>(uid);
+            RemComp<NoRotateOnMoveComponent>(uid);
             WakeNPC(uid, component);
         }
```

Comment:
This will break if they leave and go back in, better to just have combat mode toggled.